### PR TITLE
Claiming rewards in Polkadot

### LINKED ISSUE #3924: We want claim rewards in polkadot!
**Is your feature request related to a problem? Please describe.**
We are ready to start implementing claim rewards for polkadot.

**Describe the solution you'd like**
Carefully implement https://polkadot.js.org/api/substrate/extrinsics.html#payoutnominator-era-eraindex-validators-vec-t-accountid-u32 as @faboweb suggested and check PolkadotJS UI implementation for reference.

### PR DESCRIPTION
**Pull request by faboweb**
CLOSES #3924 

Claiming rewards in Polkadot.

### CODE REVIEW

**Comment by Bitcoinera** [comment]
I debugged the whole thing but still getting this every time:



🤷  Any ideas?

**Comment by mariopino** [comment]
> I debugged the whole thing but still getting this every time:
> 
> 
> 
> 🤷 Any ideas?

Yep, I'm stuck there, maybe try to test doing individual claim txs instead of using a batch, just to check, but it's my only idea by now :-/

**Comment by Bitcoinera** [comment]
> Yep, I'm stuck there, maybe try to test doing individual claim txs instead of using a batch, just to check, but it's my only idea by now :-/

Oh nice, thank you for the idea!

**Comment by mariopino** [comment]
> > Yep, I'm stuck there, maybe try to test doing individual claim txs instead of using a batch, just to check, but it's my only idea by now :-/
> 
> Oh nice, thank you for the idea!

Remember to claim the oldest era first, if by mistake you claim last era rewards you will lose all the previous payouts.

**Comment by faboweb** [comment]
api.derive.staking.stakerRewards takes forever to load :/

**Comment by faboweb** [comment]
We could store the index of the validator that you need to claim the rewards also in the rewards table. This way creating the txs would be way faster.

**Comment by faboweb** [comment]
I get the same issue though with the bad proof :/

**Comment by faboweb** [comment]
Validators are:

```validators is the list of all validators that who had exposure to during era, alongside the index of who in the clipped exposure of the validator. I.e. each element is a tuple of (validator, index of who in clipped exposure of validator).````

Clipped exposure is https://polkadot.js.org/api/substrate/storage.html#erasstakersclipped-eraindex-accountid-exposure

But using that query, I get 0 values :/

**Comment by faboweb** [comment]
Using clipped exposure I also get bad proof

**Comment by faboweb** [comment]
Inquired with the Polkadot team: https://matrix.to/#/!HzySYSaIhtyWrwiwEV:matrix.org/$1588562524305508KeIWq:matrix.org?via=matrix.parity.io&via=matrix.org&via=web3.foundation

**Comment by Bitcoinera** [comment]
> Inquired with the Polkadot team: https://matrix.to/#/!HzySYSaIhtyWrwiwEV:matrix.org/$1588562524305508KeIWq:matrix.org?via=matrix.parity.io&via=matrix.org&via=web3.foundation

"Invalid transaction payment means you do not have enough funds to pay for the transaction fee. Try increasing the balance of the caller account"

In the end we do need to pay fees in Kusama / Polkadot too?

**Comment by mariopino** [comment]
> > Inquired with the Polkadot team: https://matrix.to/#/!HzySYSaIhtyWrwiwEV:matrix.org/$1588562524305508KeIWq:matrix.org?via=matrix.parity.io&via=matrix.org&via=web3.foundation
> 
> "Invalid transaction payment means you do not have enough funds to pay for the transaction fee. Try increasing the balance of the caller account"
> 
> In the end we do need to pay fees in Kusama / Polkadot too?
> 
> But we have enough balance for this no? It doesn't make much sense to me.
> 
> I also got the Transaction: Payment error when doing `claimRewards[0]` like Mario suggested...

Yap! In Polkadot every tx have a associated fee, check i.e this event triggered by a tx:

https://polkascan.io/pre/kusama/event/2160901-4

You don't have the option to set the gas price but have the option to give a tip for faster tx processing if network is busy.

**Comment by Bitcoinera** [comment]
Now getting error message for all transactions in Polkadot 🤦  (latest develops) #3963

**Comment by faboweb** [comment]
> We could store the index of the validator that you need to claim the rewards also in the rewards table. This way creating the txs would be way faster.

Plan: store the index from the rewards script also as "rewardIndex" in the db.
Ana's transactionMetadata query delivers the ready [era, [validator, index]] array as claimableRewards if it is a polkadot network and a claimTx.

File: src/signing/networkMessages/polkadot-testnet.js [hunk 1]
```diff
@@ -42,6 +42,29 @@ export async function StakeTx(senderAddress, { to, amount }, network) {
   return await getSignMessage(senderAddress, transactions)
 }
 
+export async function ClaimRewardsTx(senderAddress) {
+  let allClaimingTxs = []
+  const api = await getAPI()
+  const stakerRewards = await api.derive.staking.stakerRewards(senderAddress)
```
Comment (faboweb) [inline src/signing/networkMessages/polkadot-testnet.js:48]: still here

Comment (mariopino) [reply src/signing/networkMessages/polkadot-testnet.js:48]: Yep, but this is the derive call to get the rewards info.

File: src/ActionModal/components/ActionModal.vue [hunk 2]
```diff
@@ -737,6 +737,7 @@ export default {
           signingType: this.selectedSignMethod,
           password: this.password
         })
+        console.log(hashResult)
```
Comment (Bitcoinera) [inline src/ActionModal/components/ActionModal.vue:740]: watch out, code leftovers!

File: src/signing/networkMessages/polkadot-testnet.js (hunk 1 above)
Comment (faboweb) [reply src/signing/networkMessages/polkadot-testnet.js:48]: we agreed to move towards using our API query for rewards in another PR

**Pull request merged**